back_end.py

```diff
@@ -0,0 +1,61 @@
+from flask import Flask
+from flask import request
+from flask import jsonify
+import json
+from mongo import Listing
+
+app = Flask(__name__)
+
+listings = {
+   'listings_list': [ ]
+}
+
+@app.route('/')
+def hello_world():
+    return 'Hello, World!'
+
+@app.route('/listings', methods=['GET', 'POST'])
+def get_listings():
+    if request.method == 'GET':
+        distance = request.args.get('distance')
+        seller = request.args.get('seller')
+        if distance:
+            listings = Listing().find_by_distance(distance)
+        elif seller:
+            listings = Listing().find_by_name(seller)
+        else:
+            listings = Listing().find_all()
+        return {'listings_list': listings}
+    elif request.method == 'POST':
+        listingToAdd = request.get_json()
+        newListing = Listing(listingToAdd)
+        newListing['price'] = float(newListing['price'])
+        newListing.save()
+        resp = jsonify(newListing), 201
+        return resp
+
+
+@app.route('/listings/sort', methods=['GET'])
+def sort_listing():
+        if request.method == 'GET':
+            sort_t = request.args.get('type')
+            if sort_t == 'distance':
+                listings = Listing().sort_by_distance()
+            elif sort_t == 'price':
+                listings = Listing().sort_by_price()
+            else:
+                listings = Listing().find_all()
+            return {'listings_list': listings}
+
+
+@app.route('/listings/buy', methods=['DElETE'])
+def buy_listing():
+    if request.method == 'DELETE':
+        listing = listings({"_id": id})
+        resp = listing.remove()
+        if resp['n'] == 1:
```

Comment:
What does 1 and 204 mean

Reply:
The 1 means in essentially is checking to see if an object was deleted from the database. If you could see the rest of the code its in an if else to see if there was a successful delete or not. Status code 204 means that no content is returned but that the request was fulfilled.